engine.js

```diff
@@ -325,6 +326,16 @@ var handleUniqueLoanNumberErrors = function(counts) {
 
     HMDAEngine.fileToJson = function(file, year, next) {
         var spec = hmdaRuleSpec.getFileSpec(year);
+
+        // If file is not an instance of a stream, make it one!
+        if (typeof file.on !== 'function') { // use duck type checking to see if file is a stream obj or not
```

Comment:
The conditional above checks to see if the file param doesn't have an `on()`. If it doesn't then we create a new readable stream and push the file into it. Finally we set file to the new stream, so that we can continue using the file param as it was originally intended.
Idea for the string to stream conversion was found here: http://stackoverflow.com/a/22085851/189884